04.bowling/bowling.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env ruby
+
+score_list = ARGV[0].split(',')
+shots = []
+score_list.each do |i|
+  if i == 'X'
+    shots << 10
+    shots << 0
+  else
+    shots << i.to_i
+  end
+end
+frame_list = []
+shots.each_slice(2) do |shot|
+  frame_list << if shot[0] == 10
+                  [shot[0]]
+                else
+                  shot
+                end
+end
+frame_list_modified = []
+if frame_list.count > 10
+  frame_sep_after_list = frame_list.slice(9..)
+  frame_sep_after_list.flatten!
+  frame_list_modified = frame_list.slice(0..8)
+  frame_list_modified << frame_sep_after_list
```

Comment:
なんだったらこうやって一行で書いてもいいかもしれませんね。
あとは、処理が少しややこしくて理解するのが若干難しく見えるので、コメントで説明を書いてあげると親切かなと思いました。

```suggestion
  # 最終フレームで3投した場合、最終フレームが二つのフレームに分かれてしまっているので一つにまとめる
  frame_list_modified = frame_list.slice(0..8) + frame_list.slice(9..).flatten!
```